aws-rds-dbcluster/src/main/java/software/amazon/rds/dbcluster/BaseHandlerStd.java

```diff
@@ -401,7 +401,7 @@ protected boolean isDBClusterDeleted(
     ) {
         try {
             fetchDBCluster(proxyClient, model);
-        } catch (DbClusterNotFoundException e) {
+        } catch (DbClusterNotFoundException | IndexOutOfBoundsException e) {
```

Comment:
I thought about making this catch at fetchDBCluster (further up the call stack), but `isDBClusterDeleted` is only called in 1 callsite and contains the blast radius to just the DeleteHandler.